geos/entrypoints_test.go

```diff
@@ -815,3 +815,63 @@ func TestMakeValid(t *testing.T) {
 		})
 	}
 }
+
+func TestCoverageUnion(t *testing.T) {
+	for i, tc := range []struct {
+		input   string
+		output  string
+		wantErr bool
+	}{
+		{
+			input: `GEOMETRYCOLLECTION(
+				POLYGON((0 0,0 1,1 0,0 0)),
+				POLYGON((1 1,0 1,1 0,1 1))
+			)`,
+			output: `POLYGON((0 0,0 1,1 1,1 0,0 0))`,
+		},
+		{
+			// Input constraint violated: inputs overlap.
+			input: `GEOMETRYCOLLECTION(
+				POLYGON((0 0,0 1,1 0,0 0)),
+				POLYGON((0 0,0 1,1 1,1 0,0 0))
+			)`,
+			wantErr: true,
+		},
+		{
+			// Input constraint violated: inputs overlap and not noded correctly.
+			input: `GEOMETRYCOLLECTION(
+				POLYGON((0 0,0 1,1 0,0 0)),
+				POLYGON((0 0,0 1,1 1,0 0))
+			)`,
+			wantErr: true,
+		},
+		{
+			// Input constraint violated: not noded correctly.
+			input: `GEOMETRYCOLLECTION(
+				POLYGON((0 0,0 1,1 1,1 0,0 0)),
+				POLYGON((0 1,2 1,2 2,0 2,0 1))
+			)`,
+			wantErr: true,
+		},
+		{
```

Comment:
Would it be useful adding another test case where only the vertex is shared?

```
		{
			// Input constraint violated: not noded correctly - shared vertex but not edge.
			input: `GEOMETRYCOLLECTION(
				POLYGON((0 0,0 1,1 1,1 0,0 0)),
				POLYGON((0 1,2 1,2 2,1 2,0 1))
			)`,
			wantErr: true,
		},
```

I was also thinking of an example where polygons have no shared nodes (I hope that's the right terminology!); put simply, they're not touching. Although, maybe the tests already somewhat imply this.

Reply:
It's actually a valid coverage if just a vertex is shared (which indicates my doc comment is probably not good enough!). I've added some more test cases in dccf76d65b282d8d2102e0b534bbd6daf550281e as well.